pkg/apiserver/authentication/authenticators/basic/basic.go

```diff
@@ -29,27 +32,48 @@ import (
 // and group from user.AllUnauthenticated. This helps requests be passed along the handler chain,
 // because some resources are public accessible.
 type basicAuthenticator struct {
-	im im.IdentityManagementInterface
+	authenticator im.PasswordAuthenticator
+	loginRecorder im.LoginRecordInterface
 }
 
-func NewBasicAuthenticator(im im.IdentityManagementInterface) authenticator.Password {
+func NewBasicAuthenticator(loginRecorder im.LoginRecordInterface, authenticator im.PasswordAuthenticator) authenticator.Password {
 	return &basicAuthenticator{
-		im: im,
+		authenticator: authenticator,
+		loginRecorder: loginRecorder,
 	}
 }
 
 func (t *basicAuthenticator) AuthenticatePassword(ctx context.Context, username, password string) (*authenticator.Response, bool, error) {
 
-	providedUser, err := t.im.Authenticate(username, password)
+	providedUser, err := t.authenticator.Authenticate(username, password)
 
 	if err != nil {
+		if err == im.AuthFailedIncorrectPassword {
```

Comment:
Just go through the process again, I don't think we need create login record for non-explict login request. Imagine this, if a user issues a lot requests within a short time, we will create lots of login records for this. For now, I think we'd better create record for explicit login request(POST /login), dealing with this later